realm/realm-library/src/main/java/io/realm/internal/TableOrView.java

```diff
@@ -373,4 +373,322 @@
      * @return the version number for the updated TableView.
      */
     long syncIfNeeded();
+
+
+    /**
+     * From the classes which are based on {@link TableView}, it makes task straight-forward to check
+     * if the originating {@link Table} is alive rather than querying to Group::has_table().
+     *
+     * @return {@code true} if the originating {@link Table} is alive, {@code false} otherwise.
+     */
+    boolean isAttached();
+
+    /**
+     * This is a void manifestation of {@link TableOrView} that takes calls and does nothing or returns
+     * {@code null} value to imitate an empty {@link TableView}.
+     */
+    TableOrView EMPTY_TABLEORVIEW = new TableOrView() {
```

Comment:
Crazy idea, but before a class is removed save a snapshot of the column names/field types/field indices. It would mean that this class would still respond to basic queries like column name/index but otherwise behave as an empty tableview.

It would also mean you could remove all `tableExist()` calls in RealmResults.

Downside is that we need to instantiate a copy of this class for each type being removed. I think that is acceptable.